### DISCUSSION TITLE
RecursiveLock deinit commented out.

### DISCUSSION THREAD

**Discussion by rsaccone** [hmlongco/Factory#201] [General]
Hello,

I am currently updating an app to use version 2.3.2 from 2.1.5 of Factory.  In reviewing the changes I noticed that in the RecursiveLock struct the deinit has been commented out and it wasn't in 2.1.5.  I am curious to learn the circumstances as to why it was commented out as part of my due diligence in bringing in the update.

Thank You,
Rob

**Comment by hmlongco** [comment]
For all practical purposes it doesn't matter, as there's one and only one globalRecursiveLock in existence for the lifetime of the application and it's only visible to Factory.

It could get reassigned IF in debug mode and IF you're running Factory unit tests and IF you force Factory into a fatalError state, but that state only exists for code coverage and would never exist occur in production code.

**Reply by rsaccone** [reply]
Thanks for the reply and explanation.  I did do a search and found the one globalRecursive lock but I didn't know about the unit test circumstances.
